# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/plasma-workspace/fullscreen-splash.diff
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Make the Plasma splash screen logo cover the whole screen and pin the
busy indicator to the lower quarter, independent of the logo geometry.

Signed-off-by: René Rebe <rene@exactco.de>

--- plasma-workspace-6.7.4/lookandfeel/org.kde.breeze/contents/splash/Splash.qml.vanilla	2026-08-05 09:35:42.074913854 +0200
+++ plasma-workspace-6.7.4/lookandfeel/org.kde.breeze/contents/splash/Splash.qml	2026-08-05 09:32:41.820627278 +0200
@@ -1,5 +1,6 @@
 /*
     SPDX-FileCopyrightText: 2014 Marco Martin <mart@kde.org>
+    SPDX-FileCopyrightText: 2026 René Rebe <rene@exactco.de>
 
     SPDX-License-Identifier: GPL-2.0-or-later
 */
@@ -31,23 +32,23 @@ Rectangle {
 
         Image {
             id: logo
-            //match SDDM/lockscreen avatar positioning
-            readonly property real size: Kirigami.Units.gridUnit * 8
 
-            anchors.centerIn: parent
+            anchors.fill: parent
 
             asynchronous: true
             source: "images/plasma.svgz"
 
-            sourceSize.width: size
-            sourceSize.height: size
+            fillMode: Image.PreserveAspectCrop
+            clip: true
+            //render the SVG at full screen resolution
+            sourceSize.width: parent.width
+            sourceSize.height: parent.height
         }
 
         // TODO: port to PlasmaComponents3.BusyIndicator
         Image {
             id: busyIndicator
-            //in the middle of the remaining space
-            y: parent.height - (parent.height - logo.y) / 2 - height/2
+            y: parent.height * 0.75 - height / 2
             anchors.horizontalCenter: parent.horizontalCenter
             asynchronous: true
             source: "images/busywidget.svgz"
